gst/nnstreamer/nnstreamer_conf.c

```diff
@@ -404,17 +419,25 @@ nnsconf_loadconf (gboolean force_reload)
 
       conf.extra_conffile =
           g_key_file_get_string (key_file, "common", "extra_config_path", NULL);
-      conf.conf[NNSCONF_PATH_FILTERS].path[CONF_SOURCE_INI] =
-          g_key_file_get_string (key_file, "filter", "filters", NULL);
-      conf.conf[NNSCONF_PATH_DECODERS].path[CONF_SOURCE_INI] =
-          g_key_file_get_string (key_file, "decoder", "decoders", NULL);
-      conf.conf[NNSCONF_PATH_CUSTOM_FILTERS].path[CONF_SOURCE_INI] =
-          g_key_file_get_string (key_file, "filter", "customfilters", NULL);
-      conf.conf[NNSCONF_PATH_CONVERTERS].path[CONF_SOURCE_INI] =
-          g_key_file_get_string (key_file, "converter", "converters", NULL);
+
+      _fill_subplugin_path (&conf, key_file, CONF_SOURCE_INI);
     }
 
     g_key_file_free (key_file);
+
+    /* load from extra config file */
+    if (conf.extra_conffile &&
+        g_file_test (conf.extra_conffile, G_FILE_TEST_IS_REGULAR)) {
+      key_file = g_key_file_new ();
+      g_assert (key_file != NULL); /** Internal lib error? out-of-memory? */
+
+      if (g_key_file_load_from_file (key_file, conf.extra_conffile,
+              G_KEY_FILE_NONE, NULL)) {
+        _fill_subplugin_path (&conf, key_file, CONF_SOURCE_EXTRA_CONF);
+      }
+
+      g_key_file_free (key_file);
+    }
```

Comment:
Create else case and handle it.
(if conf.extra_conffile == NULL, it's ok. but if conf.extra_conffile is not null and file not found, let users know what's wrong and how they can fix it.)

Reply:
Updated. Please review the changes again.